dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_thresholds.tsx

```diff
@@ -87,6 +88,9 @@ export const ConfigThresholds = ({
         fullWidth
         size="s"
         onClick={handleAddThreshold}
+        {...(props?.maxLimit && {
+          isDisabled: !isEmpty(vizState) && vizState.length === props?.maxLimit,
```

Comment:
`props?.maxLimit`, this question mark may not be required, as you have checked that already in the previous condition.